opm/simulators/wells/GroupEconomicLimitsChecker.cpp

```diff
@@ -277,16 +306,45 @@ displayDebugMessage(const std::string &msg) const
 
 void
 GroupEconomicLimitsChecker::
-closeWellsRecursive(Group group)
+addPrintMessage(const std::string &msg, const double value, const double limit, const UnitSystem::measure measure)
+{
+    const std::string header = fmt::format(
+        "{}At time = {:.2f} {} (date = {}): Group {} will close because: \n", this->message_separator,
```

Comment:
Note: Since you're already using `fmt::format()` you don't really need to store the `message_separator`.  Rather, it can be generated on the fly using libfmt's dynamic field witdth support
```C++
const auto empty_string = std::string{};
const auto num_asterisks = ...;
const std::string header = fmt::format("{:*>{}}\nAt time...", empty_string, num_asterisks, ...);
```
The `empty_string` is admittedly a bit of a hack here, but it does achieve the desired effect.

Reply:
Nice. Went for a slightly different solution to avoid a bit of code duplication.